### PR TITLE
fix: update min gas fee

### PR DESCRIPTION
**Pull request by greged93**
Set a minimum gas fee when calling estimate_gas endpoint.

Time spent on this PR: 0.1 day

Resolves: #NA

# Pull Request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?
Estimate fee works on MM even if actual estimation is under 21,000 gas.

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: crates/core/src/client/constants.rs [hunk 1]
```diff
@@ -54,6 +54,12 @@ pub mod gas {
     /// on a "first come first served" basis by the Sequencer.
     /// As a result, the priority fee is set to 0.
     pub const MAX_PRIORITY_FEE_PER_GAS: U128 = U128::ZERO;
+
+    /// The minimum gas fee for a transaction
+    ///
+    /// This minimum of 21,000 is used if the returned fee estimate is lower,
```
Comment (ClementWalter) [inline crates/core/src/client/constants.rs:60]: may be worth adding some doc https://ethereum.stackexchange.com/a/34675/81474 for the record

File: crates/core/src/client/mod.rs [hunk 2]
```diff
@@ -559,7 +556,7 @@ impl<P: Provider + Send + Sync> KakarotEthApi<P> for KakarotClient<P> {
         let data = request.data.unwrap_or_default();
 
         let tx = Transaction::Eip1559(TxEip1559 {
-            chain_id,
+            chain_id: chain_id.low_u64(),
```
Comment (Eikix) [inline crates/core/src/client/mod.rs:559]: how come request.chain_id is not the same format as TxEip1559.chain_id?

**Review by ClementWalter — COMMENTED** [review]
wondering why we need to put `--legacy` to forge while we do support EIP1559 though

File: crates/core/src/client/mod.rs (hunk 2 above)
Comment (greged93) [reply crates/core/src/client/mod.rs:559]: request is U64 and TxEip1559 is u64, not sure why

**Comment by greged93** [comment]
Do we support it in the rpc?

**Review by ClementWalter — APPROVED** [review]

**Review by ClementWalter — APPROVED** [review]

**Pull request merged**